File: CS/FormMain.cs
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Data;
using System.Windows.Forms;

namespace SelfReferenced
{
    public partial class FormMain : Form
    {
        DataSet dataSet11;

        public FormMain()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, System.EventArgs e)
        {
            dataSet11 = new DataSet();
            dataSet11.Tables.Add(GetProductsDataTable());
            dataSet11.Relations.Add("ProductsProducts", dataSet11.Tables["Products"].Columns["ID"], dataSet11.Tables["Products"].Columns["ParentID"]);
            DataView dv = new DataView(dataSet11.Tables[0]);
            dv.RowFilter = "([ParentID] Is NULL)";
            gridControl1.DataSource = dv;
            gridView1.OptionsDetail.ShowDetailTabs = false;
        }


        DataTable GetProductsDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Products";
            table.Columns.Add(new DataColumn("ID", typeof(Int32)));
            table.Columns.Add(new DataColumn("ParentID", typeof(Int32)));
            table.Columns.Add(new DataColumn("ImageIndex", typeof(int)));
            table.Columns.Add(new DataColumn("Department", typeof(string)));
            table.Columns.Add(new DataColumn("Budget", typeof(int)));
            table.Columns.Add(new DataColumn("Location", typeof(string)));
            table.Rows.Add(0, null, 0, "Departament " + 0, 1000, "Location " + 0);
            for (int i = 0; i < 20; i++)
            {
                int index = i + 1;
                table.Rows.Add(index, i % 4, i % 5, "Departament " + index, (index % 7) * 1000, "Location " + index);
            }
            return table;
        }

        private void gridView1_MasterRowGetLevelDefaultView(object sender, DevExpress.XtraGrid.Views.Grid.MasterRowGetLevelDefaultViewEventArgs e)
        {
            GridView view = new GridView(gridControl1);
            view.Assign(gridControl1.MainView, true);
            view.OptionsView.ShowGroupPanel = false;
            e.DefaultView = view;
        }
        [STAThread]
        static void Main()
        {
            Application.Run(new FormMain());
        }
    }
}

File: VB/FormMain.vb
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace SelfReferenced
    Partial Public Class FormMain
        Inherits Form

        Private dataSet11 As DataSet

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
            dataSet11 = New DataSet()
            dataSet11.Tables.Add(GetProductsDataTable())
            dataSet11.Relations.Add("ProductsProducts", dataSet11.Tables("Products").Columns("ID"), dataSet11.Tables("Products").Columns("ParentID"))
            Dim dv As New DataView(dataSet11.Tables(0))
            dv.RowFilter = "([ParentID] Is NULL)"
            gridControl1.DataSource = dv
            gridView1.OptionsDetail.ShowDetailTabs = False
        End Sub


        Private Function GetProductsDataTable() As DataTable
            Dim table As New DataTable()
            table.TableName = "Products"
            table.Columns.Add(New DataColumn("ID", GetType(Int32)))
            table.Columns.Add(New DataColumn("ParentID", GetType(Int32)))
            table.Columns.Add(New DataColumn("ImageIndex", GetType(Integer)))
            table.Columns.Add(New DataColumn("Department", GetType(String)))
            table.Columns.Add(New DataColumn("Budget", GetType(Integer)))
            table.Columns.Add(New DataColumn("Location", GetType(String)))
            table.Rows.Add(0, Nothing, 0, "Departament " & 0, 1000, "Location " & 0)
            For i As Integer = 0 To 19
                Dim index As Integer = i + 1
                table.Rows.Add(index, i Mod 4, i Mod 5, "Departament " & index, (index Mod 7) * 1000, "Location " & index)
            Next i
            Return table
        End Function

        Private Sub gridView1_MasterRowGetLevelDefaultView(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.MasterRowGetLevelDefaultViewEventArgs) Handles gridView1.MasterRowGetLevelDefaultView
            Dim view As New GridView(gridControl1)
            view.Assign(gridControl1.MainView, True)
            view.OptionsView.ShowGroupPanel = False
            e.DefaultView = view
        End Sub
        <STAThread> _
        Shared Sub Main()
            Application.Run(New FormMain())
        End Sub
    End Class
End Namespace